### DISCUSSION TITLE
What's the difference getStaticPaths fallback true or false when fetching dynamic id

### DISCUSSION THREAD

**Discussion by Vince-vegas** [vercel/next.js#17607] [Help] [answered]
Which one of this code are better implementation when fetching data
Which one is fast on requesting the data

// pages/users/[id].js

**1.**
// falback is **TRUE**
```
 const UserId = ({ data }) => {
  const router = useRouter();

  if (router.isFallback) {
    return <h1>Loading</h1>;
  }

  if (Object.keys(data).length === 0) {
    return <h1>404 Page</h1>;
  }

  const { id, name, email, address } = data;
  return (
    <div>
      <Card id={id} name={name} email={email} address={address} />
    </div>
  );
};

export async function getStaticPaths() {
  return {
    paths: [],
    fallback: true,
  };
}


export async function getStaticProps(ctx) {
  const res = await fetch(
    `https://......./users/${ctx.params.id}`
  );
  const data = await res.json();

  return {
    props: {
      data,
    },
  };
}
```
 
**2.**
// fallback is **FALSE**
```
const UserId = ({ data }) => {
  const { id, name, email, address } = data;
  return (
    <div>
      <Card id={id} name={name} email={email} address={address} />
    </div>
  );
};

export async function getStaticPaths() {
  const res = await fetch('https://......./users/');
  const data = await res.json();

  const paths = data.map((item) => ({ params: { id: `${item.id}` } }));

  return {
    paths,
    fallback: false,
  };
}

export async function getStaticProps(ctx) {
  const res = await fetch(
    `https:/....../users/${ctx.params.id}`
  );
  const data = await res.json();

  return {
    props: {
      data,
    },
  };
}
```

**Comment by lfades** [comment, marked as answer]
Both are equally fast, they differ on this:

`fallback: false`: "I only want you to pre-render the pages that are defined in `getStaticPaths`, if someone tries to go to a different path, send a 404!"

`fallback: true`: "I want you to pre-render the pages that are defined in `getStaticPaths`, and also a fallback page, so if someone visits a path that wasn't defined in `getStaticPaths` you'll use the fallback page, once the fallback page loads you'll generate the data and cache it, so future requests to the same path don't require the usage of the fallback page again!"

For more details visit our [`fallback` docs](https://nextjs.org/docs/basic-features/data-fetching#the-fallback-key-required)!
